metadata/src/main/java/org/apache/kafka/controller/QuorumController.java

```diff
@@ -1271,6 +1271,9 @@ private void replay(ApiMessage message, Optional<OffsetAndEpoch> snapshotId, lon
                 case PRODUCER_IDS_RECORD:
                     producerIdControlManager.replay((ProducerIdsRecord) message);
                     break;
+                case BROKER_REGISTRATION_CHANGE_RECORD:
+                    clusterControl.replay((BrokerRegistrationChangeRecord) message);
+                    break;
```

Comment:
@cmccabe I think you missed this comment:
> Don't we also need broker side changes. At a minimum don't we need to change MetadataDelta#replay? Do we need an integration test that shows that controllers and brokers support this record?

from https://github.com/apache/kafka/pull/12195#pullrequestreview-985201342